File: packages/swc-plugin/src/visitor/analysis.rs
use swc_core::ecma::atoms::Atom;

/// Check if a component name matches known gt-next translation components
pub fn is_translation_component_name(name: &Atom) -> bool {
  matches!(name.as_ref(), "T")
}

/// Check if a component name matches known gt-next variable components
pub fn is_variable_component_name(name: &Atom) -> bool {
  matches!(name.as_ref(), "Var" | "Num" | "Currency" | "DateTime")
}

/// Check if a name is a GT branch
pub fn is_branch_name(name: &Atom) -> bool {
  matches!(name.as_ref(), "Branch" | "Plural")
}

/// Check if a name is a GT translation function
pub fn is_translation_function_name(name: &Atom) -> bool {
  matches!(name.as_ref(), "useGT" | "getGT")
}

File: packages/babel/src/visitor/analysis.ts
/**
 * Analysis utilities for identifying GT components and functions
 */

/**
 * Check if a component name matches known gt-next translation components
 */
export function isTranslationComponent(name: string): boolean {
  return name === 'T';
}

/**
 * Check if a component name matches known gt-next variable components
 */
export function isVariableComponent(name: string): boolean {
  return ['Var', 'Num', 'Currency', 'DateTime'].includes(name);
}

/**
 * Check if a name is a GT branch component
 */
export function isBranchComponent(name: string): boolean {
  return ['Branch', 'Plural'].includes(name);
}

/**
 * Check if a name is a GT translation function
 */
export function isTranslationFunction(name: string): boolean {
  return ['useGT', 'getGT'].includes(name);
}

/**
 * Check if it's a translation function callback (const t = useGT())
 */
export function isTranslationFunctionCallback(name: string): boolean {
  return ['useGT_callback', 'getGT_callback'].includes(name);
}